internal/server/iigointernal/orchestration.go

```diff
@@ -186,42 +212,70 @@ func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.C
 	if !CheckEnoughInCommonPool(costOfElection, g) {
 		return false, "Insufficient budget to run IIGO elections"
 	}
-	g.JudgeID, appointJudgeError = legislativeBranch.appointNextJudge(judgeMonitored, g.JudgeID, aliveClientIds)
+	appointedJudge, appointJudgeError := legislativeBranch.appointNextJudge(judgeMonitored, g.JudgeID, aliveClientIds)
 	if appointJudgeError != nil {
 		return false, "Judge was not apointed by the Speaker. Insufficient budget"
 	}
 	// Get new Speaker ID
-	g.SpeakerID, appointSpeakerError = executiveBranch.appointNextSpeaker(speakerMonitored, g.SpeakerID, aliveClientIds)
+	appointedSpeaker, appointSpeakerError := executiveBranch.appointNextSpeaker(speakerMonitored, g.SpeakerID, aliveClientIds)
 	if appointSpeakerError != nil {
 		return false, "Speaker was not apointed by the President. Insufficient budget"
 	}
 	// Get new President ID
-	g.PresidentID, appointPresidentError = judicialBranch.appointNextPresident(presidentMonitored, g.PresidentID, aliveClientIds)
+	appointedPresident, appointPresidentError := judicialBranch.appointNextPresident(presidentMonitored, g.PresidentID, aliveClientIds)
 	if appointPresidentError != nil {
 		return false, "President was not apointed by the Judge. Insufficient budget"
 	}
 
-	// Pay salaries into budgets
-	errorJudicial := judicialBranch.sendPresidentSalary()
-	errorLegislative := legislativeBranch.sendJudgeSalary()
-	errorExecutive := executiveBranch.sendSpeakerSalary()
-	// Return false only after attempting to pay all roles their salary
-	if errorJudicial != nil || errorLegislative != nil || errorExecutive != nil {
-		return false, "Cannot pay IIGO salary"
+	//Monitor again for election fraud
+	speakerMonitored = monitoring.monitorRole(g, iigoClients[g.JudgeID])
+	presidentMonitored = monitoring.monitorRole(g, iigoClients[g.SpeakerID])
+	judgeMonitored = monitoring.monitorRole(g, iigoClients[g.PresidentID])
+
+	// Get new Judge ID
+	actionCost = gameConf.IIGOConfig
+	costOfElection = actionCost.AppointNextSpeakerActionCost + actionCost.AppointNextJudgeActionCost + actionCost.AppointNextPresidentActionCost
+	if !CheckEnoughInCommonPool(costOfElection, g) {
+		return false, "Insufficient budget to run IIGO elections"
+	}
+	g.JudgeID, appointJudgeError = legislativeBranch.appointNextJudge(judgeMonitored, appointedJudge, aliveClientIds)
+	if appointJudgeError != nil {
+		return false, "Judge was not apointed by the Speaker. Insufficient budget"
+	}
+	// Get new Speaker ID
+	g.SpeakerID, appointSpeakerError = executiveBranch.appointNextSpeaker(speakerMonitored, appointedSpeaker, aliveClientIds)
+	if appointSpeakerError != nil {
+		return false, "Speaker was not apointed by the President. Insufficient budget"
+	}
+	// Get new President ID
+	g.PresidentID, appointPresidentError = judicialBranch.appointNextPresident(presidentMonitored, appointedPresident, aliveClientIds)
+	if appointPresidentError != nil {
+		return false, "President was not apointed by the Judge. Insufficient budget"
 	}
```

Comment:
Why are we monitoring twice and running elections twice?

Reply:
Because we have to monitor elections, otherwise any role has nothing stopping it from putting whoever it wants, self-included, into power. But elections depend on monitoring results. Hence monitor->elect->monitor->reelect

Reply:
Should we increase the salary to accommodate having 2 sets of monitoring?

Reply:
Agents decide the salary and budget. But we can decrease the cost of monitoring